src/Std/Time.lean

```diff
@@ -0,0 +1,173 @@
+- **`ZonedDateTime`**: An existential version of the `DateTime`.
+
+## Duration
+Represents spans of time and the difference between two points in time.
+
+- **`Duration`**: Represents the time span or difference between two `Timestamp`s values with nanosecond precision.
+
+# Formats
+
+Format strings are used to convert between `String` representations and date/time types, e.g, `YYYY-MM-DD'T'hh:mm:ss.sssZ`.
+The supported formats include:
+
+- **Year:**
+  - `YYYY`: Four-digit year (e.g., 2024)
+  - `YY`: Two-digit year (e.g., 24 for 2024)
+- **Month:**
+  - `MMMM`: Full month name (e.g., January, February)
+  - `MMM`: Abbreviated month name (e.g., Jan, Feb)
+  - `MM`: Two-digit month (e.g., 01 for January)
+  - `M`: One or two-digit month (e.g., 1 for January, 10 for October)
+- **Day:**
+  - `DD`: Two-digit day of the month (e.g., 01, 02)
+  - `D`: One or two-digit day of the month (e.g., 1, 2)
+  - `d`: One or two-digit day of the month with space padding (e.g., " 1", "12")
+- **Day of Week:**
+  - `EEEE`: Full name of the day of the week (e.g., Monday, Tuesday)
+  - `EEE`: Abbreviated day of the week (e.g., Mon, Tue)
+- **Hour:**
+  - `hh`: Two-digit hour in 24-hour format (e.g., 13, 14)
+  - `h`: One or two-digit hour in 24-hour format (e.g., 1, 2)
+  - `HH`: Two-digit hour in 12-hour format (e.g., 01, 02)
+  - `H`: One or two-digit hour in 12-hour format (e.g., 1, 2)
+- **AM/PM Indicator:**
+  - `AA`: Uppercase AM/PM (e.g., AM, PM)
+  - `aa`: Lowercase am/pm (e.g., am, pm)
+- **Minute:**
+  - `mm`: Two-digit minute (e.g., 01, 02)
+  - `m`: One or two-digit minute (e.g., 1, 2)
+- **Second:**
+  - `sssssssss` : Nine-digit nanosseconds (e.g., 000000009)
+  - `sss`: Three-digit milliseconds (e.g., 001, 202)
+  - `ss`: Two-digit second (e.g., 01, 02)
+  - `s`: One or two-digit second (e.g., 1, 2)
+- **Timezone:**
+  - `ZZZZZ`: Full timezone offset with hours and minutes (e.g., +03:00)
+  - `ZZZZ`: Timezone offset without the colon (e.g., +0300)
+  - `ZZZ`: Like `ZZZZ`, but with "UTC" for UTC
+  - `Z`: Like `ZZZZZ`, but with "Z" for UTC
+  - `z`: Timezone name (e.g., Brasilia Standard Time)
+
+# Macros
+
+In order to help the user build dates easily, there are a lot of macros available for creating dates.
+The `.sssssssss` can be ommited in most cases.
+
+- **`date% YYYY-MM-DD`**: Defines a date in the `YYYY-MM-DD` format.
+- **`date% HH:mm:ss:sssssssss`**: Defines a time in the `HH:mm:ss:sssssssss` format, including fractional seconds.
+- **`date% YYYY-MM-DD:HH:mm:ss.sssssssss`**: Defines a datetime in the `YYYY-MM-DD:HH:mm:ss:sssssssss` format.
+- **`date% YYYY-MM-DDTHH:mm:ss:sssssssssZ`**: Defines a datetime with a timezone in the `YYYY-MM-DDTHH:mm:ss:sssssssssZ` format.
+- **`time% HH:mm:ss:sssssssss`**: Defines a time in the `HH:mm:ss` format.
+- **`offset% +HH:mm`**: Defines a timezone offset in the format `+HH:mm`.
+- **`timezone% NAME/ID offset% +HH:mm`**: Defines a timezone with a name and an offset.
```

Comment:
Above you say that we are following ISO standards (again presumably 8601) but ISO 8601 does not have a timezone specification. I assume that these are IANA time zone identifiers?

Reply:
Yes, I'm using a format that is similar to the IANA Timezone identifiers.